Library/Homebrew/version.rb

```diff
@@ -122,20 +138,22 @@ def inspect
   class StringToken < Token
     PATTERN = /[a-z]+/i.freeze
 
+    sig { params(value: T.untyped).void }
     def initialize(value)
       super
 
       @value = value.to_s
     end
 
+    sig { override.params(other: T.untyped).returns(T.nilable(Integer)) }
     def <=>(other)
       return unless other = Token.from(other)
 
       case other
       when StringToken
         value <=> other.value
       when NumericToken, NullToken
-        -Integer(other <=> self)
+        -T.must(other <=> self)
```

Comment:
Shouldn't this be `nil`, since it's not comparable?

Reply:
No; `other <=> self` means `other.<=>(self)` which calls `NumericToken#<=>` (or `NullToken#<=>`)

Reply:
Then I guess my question is: Shouldn't `NullToken#<=>` always return `nil`?

Reply:
No, since `NullToken` is used for e.g. a comparison between `1.5` and `1.5.1` where `Version.new("1.5").patch` returns `nil`, which becomes `NULL_TOKEN` in `<=>` after calling `Token.from`

Reply:
Ah, I see.